bindings/rust/s2n-tls/src/testing.rs

```diff
@@ -60,35 +60,59 @@ impl Default for Counter {
 }
 
 pub struct CertKeyPair {
-    cert_path: &'static str,
-    cert: &'static [u8],
-    key: &'static [u8],
+    cert_path: String,
+    key_path: String,
+    ca_path: String,
+    cert: Vec<u8>,
+    key: Vec<u8>,
 }
 
 impl Default for CertKeyPair {
     fn default() -> Self {
-        CertKeyPair {
-            cert_path: concat!(
-                env!("CARGO_MANIFEST_DIR"),
-                "/../../../tests/pems/rsa_4096_sha512_client_cert.pem",
-            ),
-            cert: &include_bytes!("../../../../tests/pems/rsa_4096_sha512_client_cert.pem")[..],
-            key: &include_bytes!("../../../../tests/pems/rsa_4096_sha512_client_key.pem")[..],
-        }
+        let prefix = concat!(
+            env!("CARGO_MANIFEST_DIR"),
+            "/../../../tests/pems/rsa_4096_sha512_client_"
+        );
+        Self::from(prefix, "cert", "key", "cert")
     }
 }
 
 impl CertKeyPair {
-    pub fn cert_path(&self) -> &'static str {
-        self.cert_path
+    pub fn from(prefix: &str, chain: &str, key: &str, ca: &str) -> Self {
+        let cert_path = format!("{prefix}{chain}.pem");
+        let key_path = format!("{prefix}{key}.pem");
+        let ca_path = format!("{prefix}{ca}.pem");
+        let cert = std::fs::read(&cert_path)
+            .unwrap_or_else(|_| panic!("Failed to read cert at {cert_path}"));
+        let key =
+            std::fs::read(&key_path).unwrap_or_else(|_| panic!("Failed to read key at {key_path}"));
+        CertKeyPair {
+            cert_path,
+            key_path,
+            ca_path,
+            cert,
+            key,
+        }
     }
 
-    pub fn cert(&self) -> &'static [u8] {
-        self.cert
+    pub fn cert_path(&self) -> &str {
```

Comment:
Nit: Doesn't really matter, but since this is a test struct with no specific invariants to maintain, I'd vote for just exposing all of the `CertKeyPair` members (making them public) so we don't have to deal with the getter boilerplate.

Reply:
I prefer the getters, especially since they already exist. They already saved me from needing to update a bunch of references when I changed the implementation. Removing them would also involve touching more code unrelated to renegotiate for a trivial reason.